packages/test/src/test-workflows.ts

```diff
@@ -1706,3 +1706,28 @@ test('conditionWaiter', async (t) => {
     compareCompletion(t, completion, makeSuccess([makeCompleteWorkflowExecution()]));
   }
 });
+
+test('conditionRacer', async (t) => {
+  const { workflowType } = t.context;
+  {
+    const completion = await activate(t, makeStartWorkflow(workflowType));
+    compareCompletion(
+      t,
+      completion,
+      makeSuccess([makeStartTimerCommand({ seq: 1, startToFireTimeout: msToTs('1s') })])
+    );
+  }
+  {
+    const completion = await activate(
+      t,
+      makeActivation(
+        Date.now(),
+        {
+          signalWorkflow: { signalName: 'unblock', input: [] },
+        },
+        makeFireTimerJob(1)
+      )
+    );
+    compareCompletion(t, completion, makeSuccess([{ cancelTimer: { seq: 1 } }]));
```

Comment:
How did you make this repro w/o the extra activation?

Reply:
"extra"? I think w/ these two jobs:

```
        {
          signalWorkflow: { signalName: 'unblock', input: [] },
        },
        makeFireTimerJob(1)
```

It processes the first, which results in a cancel (and immediate cleanup/removal of timer completion), and then when it tries to process the second, it drops it because it can't find the completion (and assumes it must have been canceled). (or in the repro case, throws)

Reply:
The issue I had initially is that the fireTimer job doesn't get delivered if the workflow is already complete.
Adding another condition at the end of the workflow prevents that.